decisions/0015-manage-short-lived-access-tokens-in-tests.md

```diff
@@ -0,0 +1,36 @@
+# 15. Manage short-lived access tokens in tests
+
+Date: 2019-01-16
+
+## Status
+
+Accepted
+
+## Context
+
+We have implemented the VCR gem as part of our testing suite; this gem allows you to "record your test suite's HTTP interactions and replay them during future test runs for fast, deterministic, accurate tests."
+
+VCR allows you to filter out sensitive information, but this does not cover any data returned in the body.  We get a short-lived access token returned in the body as part of the OAuth flow and need to find a way of ensuring that these are not exposed and used to get access to sensitive data.  Whilst the tests are run against T3, which is an anonymised dataset, we don't want to risk the possibility of someone being able to engineer a way of undoing that and seeing real data.
+
+The main options:
+
+- making the github repository private
+- git-crypting the VCR cassettes folder
+- adding the VCR cassettes folder to .gitignore
+
+Making the github repository private, even for a very short period of time, as we are committed to coding in the open and it is very difficult to open up a repository once it has been made private.
```

Comment:
Making the github repository private contradicts our commitment to coding in the open, and eventually opening a previously closed repository is often risky and hard to prioritise.